browser/ui/views/permissions/permission_prompt_bubble_base_view_browsertest.cc

```diff
@@ -0,0 +1,112 @@
+            std::make_unique<permissions::ContentSettingPermissionResolver>(
+                ContentSettingsType::MEDIASTREAM_CAMERA),
+            true, GURL()),
+        base::DoNothing()));
+  }
+  ~MockPermissionPromptDelegate() override = default;
+
+  const std::vector<std::unique_ptr<permissions::PermissionRequest>>& Requests()
+      override {
+    return request_;
+  }
+
+  GURL GetRequestingOrigin() const override { return GURL(); }
+
+  GURL GetEmbeddingOrigin() const override { return GURL(); }
+
+  void Accept() override {}
+  void AcceptThisTime() override {}
+  void Deny() override {}
+  void Dismiss() override {}
+  void Ignore() override {}
+  void FinalizeCurrentRequests() override {}
+  void OpenHelpCenterLink(const ui::Event& event) override {}
+  void PreIgnoreQuietPrompt() override {}
+  void SetManageClicked() override {}
+  void SetLearnMoreClicked() override {}
+  void SetHatsShownCallback(base::OnceCallback<void()> callback) override {}
+
+  bool WasCurrentRequestAlreadyDisplayed() override { return false; }
+  bool ShouldDropCurrentRequestIfCannotShowQuietly() const override {
+    return false;
+  }
+  bool ShouldCurrentRequestUseQuietUI() const override { return false; }
+  std::optional<permissions::PermissionUiSelector::QuietUiReason>
+  ReasonForUsingQuietUi() const override {
+    return std::nullopt;
+  }
+  void SetDismissOnTabClose() override {}
+  void SetPromptShown() override {}
+  void SetDecisionTime() override {}
+  bool RecreateView() override { return false; }
+  const permissions::PermissionPrompt* GetCurrentPrompt() const override {
+    return nullptr;
+  }
+
+  base::WeakPtr<permissions::PermissionPrompt::Delegate> GetWeakPtr() override {
+    return weak_ptr_factory_.GetWeakPtr();
+  }
+
+  content::WebContents* GetAssociatedWebContents() override { return nullptr; }
+
+ private:
+  std::vector<std::unique_ptr<permissions::PermissionRequest>> request_;
+
+  base::WeakPtrFactory<MockPermissionPromptDelegate> weak_ptr_factory_{this};
+};
+
+class MockPermissionPromptBubbleBaseView
+    : public PermissionPromptBubbleBaseView {
+ public:
+  MockPermissionPromptBubbleBaseView(
+      Browser* browser,
+      base::WeakPtr<permissions::PermissionPrompt::Delegate> delegate)
+      : PermissionPromptBubbleBaseView(browser,
+                                       delegate,
+                                       base::TimeTicks::Now(),
+                                       PermissionPromptStyle::kBubbleOnly) {
+    CreateWidget();
+    ShowWidget();
+  }
+};
+
+}  // namespace
+
+IN_PROC_BROWSER_TEST_F(PermissionPromptBubbleBaseViewBrowserTest,
+                       ZOrderLevelShouldBeSecuritySurface) {
+  // This test checks that the permission prompt bubble is created with the
+  // correct z-order level, which should be kSecuritySurface.
+  MockPermissionPromptDelegate mock_delegate;
+  auto* permission_prompt = new MockPermissionPromptBubbleBaseView(
```

Comment:
Another thing I missed here: Does this object just deletes itself on `CloseNow`?

Reply:
Yes, I believe it'd be deleted when native handle of widget is deleted.